staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go

```diff
@@ -195,10 +203,38 @@ func (c *namedCreaterAdapter) Create(ctx context.Context, name string, obj runti
 	return c.Creater.Create(ctx, obj, createValidatingAdmission, options)
 }
 
-// manager is assumed to be already a valid value, we need to make
-// userAgent into a valid value too.
-func managerOrUserAgent(manager, userAgent string) string {
-	if manager != "" {
+// validFieldManager takes the fieldManager from the passed in object and option to validate
+// the option and object fieldManager match if both are set
+func validFieldManager(obj runtime.Object, option string) (string, error) {
+	m, err := meta.Accessor(obj)
+	if err != nil {
+		return "", fmt.Errorf("couldn't get accessor: %v", err)
+	}
+
+	manager := m.GetFieldManager()
+	if len(manager) != 0 {
+		if errs := validation.ValidateFieldManager(manager, field.NewPath("fieldManager", "metadata")); len(errs) > 0 {
+			return "", errors.NewInvalid(schema.GroupKind{Group: metav1.GroupName, Kind: "ObjectMeta"}, "", errs)
+		}
+	}
+
+	if len(option) != 0 {
+		if len(manager) != 0 && option != manager {
+			return "", errors.NewBadRequest("the fieldManager of the provided object does not match the fieldManager sent on the request")
+		}
+
+		return option, nil
+	}
+
+	if len(manager) != 0 {
```

Comment:
Nit, but I suspect that this condition is pointless, because `return manager, nil` would return the same thing anyway.